lib/streamlit/runtime/state/session_state.py

```diff
@@ -280,6 +280,50 @@ def _missing_key_error_message(key: str) -> str:
     )
 
 
+@dataclass
+class KeyIdMapper:
+    """A mapping of user-provided keys to widget IDs.
+    It also maps widget IDs to user-provided keys so that this reverse mapping
```

Comment:
nit: maybe call it element ID. Since we also have elements that don't have widget-like capabilities with IDs, I'm trying to unify around this name instead to lower confusions.

Reply:
I updated it 👍